### PR TITLE
Feature/dhfprod 1708 run ingest

### PR DESCRIPTION
**Pull request by aebadirad**

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs [hunk 1]
```diff
@@ -254,10 +258,11 @@ class Flow {
...
       }
-      for (let uri of uris) {
-        flowInstance.globalContext.uri = uri;
-        let result = flowInstance.runMain(uri, content[uri], combinedOptions, processor.run);
-        flowInstance.addToWriteQueue(uri, result, flowInstance.globalContext);
+      for (let contentItem of content) {
+        flowInstance.globalContext.uri = contentItem.uri;
+        let result = flowInstance.runMain(contentItem, combinedOptions, processor.run);
+        flowInstance.addToWriteQueue(result, flowInstance.globalContext);
+        this.putMetadata(contentItem.uri, flowName, step.name);
```
Comment (hao1st) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs]: the line should be removed...

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs [hunk 2]
```diff
@@ -328,6 +352,114 @@ class FlowUtils {
...
+    let headers = {};
+    for (let key in options.headers) {
+      headers[key] = this.evalSubstituteVal(options.headers[key]);
+    }
+    return headers;
+  }
+
+  createMetadata(metaData = {}, flowName, stepName) {
+    metaData[this.consts.CREATED_ON] = this.datahub.flowUtils.evalSubstituteVal(this.consts.CREATED_ON);
+    metaData[this.consts.CREATED_BY] = this.datahubflowUtils.evalSubstituteVal(this.consts.CREATED_BY);
```
Comment (hao1st) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs]: https://github.com/marklogic/marklogic-data-hub/blob/da4e0f04fbda0b01cb3998274374d41bb9b2297b/marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs#L456-L457

metaData[this.consts.CREATED_ON] = this.evalSubstituteVal(this.consts.CREATED_ON);
metaData[this.consts.CREATED_BY] = this.evalSubstituteVal(this.consts.CREATED_BY);

Comment (aebadirad) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs]: Done.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs [hunk 3]
```diff
@@ -254,10 +259,15 @@ class Flow {
...
-        flowInstance.globalContext.uri = uri;
-        let result = flowInstance.runMain(uri, content[uri], combinedOptions, processor.run);
-        flowInstance.addToWriteQueue(uri, result, flowInstance.globalContext);
+      let self = this;
+      for (let contentItem of content) {
+        flowInstance.globalContext.uri = contentItem.uri;
+        let result = flowInstance.runMain(contentItem, combinedOptions, processor.run);
+        //add our metadata to this
+        if(result && result.context) {
+          result.context.metadata = self.flowUtils.createMetadata( result.context ? result.context.metadata : {}, flowName, step.name);
```
Comment (akshaysonvane) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs]: Should this be
```suggestion
          result.context.metadata = self.flowUtils.createMetadata( result.context.metadata ? result.context.metadata : {}, flowName, step.name);
```

**Comment by aebadirad** [comment]
We'll have to keep the context deletions so we don't return that back to MLCP (because mlcp will write it) and update the MLCP context object (which is different than the context object we built).

**Pull request merged**